src/core/entities/crux-user.ts

```diff
@@ -0,0 +1,70 @@
+import { CruxId } from "src/packages/identity-utils";
+import { BaseError } from "../../packages/error";
+import { getLogger } from "../../packages/logger";
+import { CruxSpec } from "./crux-spec";
+
+const log = getLogger(__filename);
+
+export interface IAddress {
+    addressHash: string;
+    secIdentifier?: string;
+}
+
+export interface IAddressMapping {
+    [currency: string]: IAddress;
+}
+
+export interface ICruxUserRegistrationStatus {
+    status: SubdomainRegistrationStatus;
+    statusDetail: SubdomainRegistrationStatusDetail;
+}
+
+export enum SubdomainRegistrationStatus {
+    NONE = "NONE",
+    PENDING = "PENDING",
+    DONE = "DONE",
+    REJECT = "REJECT",
+}
+
+export enum SubdomainRegistrationStatusDetail {
+    NONE = "Subdomain not registered with this registrar.",
+    PENDING_REGISTRAR = "Subdomain registration pending on registrar.",
+    PENDING_BLOCKCHAIN = "Subdomain registration pending on blockchain.",
+    DONE = "Subdomain propagated.",
+}
+
+export class CruxUser {
+    public cruxID: CruxId;
+    public registrationStatus: ICruxUserRegistrationStatus;
+    private addressMap: IAddressMapping;
+
+    constructor(cruxID: CruxId, addressMap: IAddressMapping, registrationStatus: ICruxUserRegistrationStatus) {
+        this.cruxID = cruxID;
+        this.addressMap = addressMap;
+        this.registrationStatus = this.setRegistrationStatus(registrationStatus);
+    }
+    public getAddressMap(): IAddressMapping {
+        return this.addressMap;
+    }
+    public setAddressMap(addressMap: IAddressMapping) {
+        try {
+            CruxSpec.validations.validateAddressMap(addressMap);
+        } catch (error) {
+            throw new BaseError(error, `Address Map validation failed!`);
+        }
+        this.addressMap = addressMap;
+    }
+    public getAddressFromAsset(assetId: string): IAddress {
+        return this.addressMap[assetId] || this.addressMap[assetId.toLowerCase()];
+    }
+    private setRegistrationStatus = (registrationStatus: ICruxUserRegistrationStatus) => {
+        // validate and set the registrationStatus
+        if (!(Object.values(SubdomainRegistrationStatus).includes(registrationStatus.status))) {
+            throw new BaseError(null, `Subdomain registration status validation failed!`);
+        }
+        if (!(Object.values(SubdomainRegistrationStatusDetail).includes(registrationStatus.statusDetail))) {
+            throw new BaseError(null, `Subdomain registration status detail validation failed!`);
+        }
+        return registrationStatus;
+    }
+}
```

Comment:
can be supplemented with getter/setters:
```
const user = new CruxUser();
// getting address map
const userAddressMap = user.addressMap;

// setting address map
const newAddressMap = {};
user.addressMap = newAddressMap;
await userRepo().save(user);
```

Reply:
Getters and setters were there and were removed as per Ankit's suggestion